lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -896,6 +896,179 @@ class ConvertAtenUpsampleNearest2dVecOp
+    SmallVector<Value, 2> scaleFactorsFloatValues;
+
+    SmallVector<Value, 4> inputSizeTorchInt;
+    if (!getListConstructElements(op.input_size(), inputSizeTorchInt))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: the input_size is not constructed from "
+              "ListConstruct");
+    SmallVector<Value, 4> inputSizeIntValues;
+    inputSizeIntValues = getTypeConvertedValues(
+        rewriter, loc, getTypeConverter(), inputSizeTorchInt);
+
+    // The dimension at which the scaling starts.
+    unsigned hDimOffset = 2;
+
+    if (!op.scale_factors().getType().isa<Torch::NoneType>()) {
+      SmallVector<Value, 2> scaleFactorsTorchFloat;
+      if (!getListConstructElements(op.scale_factors(), scaleFactorsTorchFloat))
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: the scale_factors is not constructed from "
+                "ListConstruct");
+      scaleFactorsFloatValues = getTypeConvertedValues(
+          rewriter, loc, getTypeConverter(), scaleFactorsTorchFloat);
+      for (unsigned i = 0; i < scaleFactorsFloatValues.size(); i++)
+        scaleFactorsFloatValues[i] =
+            convertScalarToDtype(rewriter, loc, scaleFactorsFloatValues[i],
+                                 mlir::Float32Type::get(op->getContext()));
+    } else {
+      for (unsigned i = 0; i < 2; i++) {
+        auto scaleFactorVal = rewriter.create<arith::DivFOp>(
+            loc,
+            convertScalarToDtype(rewriter, loc,
+                                 gradOutputSizeIntValues[hDimOffset + i],
+                                 mlir::Float32Type::get(op->getContext())),
+            convertScalarToDtype(rewriter, loc,
+                                 inputSizeIntValues[hDimOffset + i],
+                                 mlir::Float32Type::get(op->getContext())));
+        scaleFactorsFloatValues.push_back(scaleFactorVal);
+      }
+    }
+
+    SmallVector<Value, 2> scaleFactorsIntValues;
+    for (auto v : scaleFactorsFloatValues)
+      scaleFactorsIntValues.push_back(convertScalarToDtype(
+          rewriter, loc, rewriter.create<math::CeilOp>(loc, v),
+          mlir::IntegerType::get(op->getContext(), 64)));
+
+    Value outTensor = createZeroInitTensor(
+        rewriter, loc,
+        castIntVectorToIndexVector(rewriter, loc, inputSizeIntValues),
+        elementType);
+
+    Value kernelTensor = rewriter.create<tensor::EmptyOp>(
+        loc,
+        getAsOpFoldResult(
+            castIntVectorToIndexVector(rewriter, loc, scaleFactorsIntValues)),
+        elementType);
+
+    SmallVector<AffineExpr> affineExprs;
+    for (unsigned i = 0; i < gradOutputRank; i++)
+      affineExprs.push_back(rewriter.getAffineDimExpr(i));
+
+    AffineMap inputMap = AffineMap::get(gradOutputRank + 2, /*symbolCount=*/0,
+                                        affineExprs, op->getContext());
+
+    affineExprs.clear();
+    for (unsigned i = 0; i < 2; i++)
```

Comment:
Try to avoid hardcoding constants. You should do `i < scaleFactorsIntValues.size()`